### PR TITLE
v4.2 Permissions Update

### PR DESCRIPTION
**Pull request by binaryoverload**

### CODE REVIEW

File: permissions.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
 # Permissions
 
-<!-- > ### In 4.2 permissions are changing! Please read the `Permissions (4.2)` page for more info! -->
+## Version 4.2 permissions changes
+In version 4.2 we have changed the permissions system to be, we believe, better. Here are some changes:
+* There is no longer a default group. We have replaced this with a dynamic system that means that users get default permissions without a group. To list all of the default permissions, please refer to the `_perms list` command!
+* We have introduced negation permissions meaning by putting a `-` before a permission will **deny** users access to the permission. This works in terms of group hierarchy; groups at the top will override groups at the bottom. User permissions will also immediately override group permissions.
```
Comment (weeryan17) [inline permissions.md:6]: As we allready discussed make sure you specify spaces arn't allowed

File: permissions.md [hunk 2]
```diff
@@ -16,7 +23,8 @@ If you wish to get going quickly these commands will be good to know!<br/>
 `_perms group <name> link <role>` - This is how you can link a group to a specific Discord role. More info [here](#group-linking)<br/>
 
 ## Groups
```
Comment (weeryan17) [inline permissions.md:25]: You should probably add move stuff here. also specify that it's order from top to bottom stating with 1

**Review by weeryan17 — CHANGES_REQUESTED** [review]
This is what i found that we probably should change

File: permissions.md (hunk 2 above)
Comment (weeryan17) [reply permissions.md:25]: Btw I'm refering to the groups section. github doesn't really make that clear

File: permissions.md [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+* We have finally added the ability to do full and accurate permission validation meaning you will never be mistake on a permission again!
+
 
 ## Important Notes
-* All FlareBot permissions follow a pattern of `flarebot.command` and `flarebot.command.sub-command` if applicable. 
+* All FlareBot permissions follow a pattern of `flarebot.command` and `flarebot.command.sub-command` if applicable.
+* All the permissions can be listed with `_perms list`. This command also tells you if the permissions are default or not.
 * Everyone with `Administrator` Discord permission will have access to **all** commands.
-* Everyone already has a group before you do anything, the group is called `Default`.
+* Everyone already has a select group of permissions to start off with. These are listed in the "Default" section of `_perms list`! To remove these permissions simply add the permission with a `-` before it to a user! (There should be no space between the `-` and the permission i.e. `-flarebot.stop`)
```
Comment (WalshyDev) [inline permissions.md]: You said you can ass this to a user but not a group.
Also stop isn't default, use something like `play` as the example.

File: permissions.md [hunk 4]
```diff
@@ -16,7 +23,8 @@ If you wish to get going quickly these commands will be good to know!<br/>
 `_perms group <name> link <role>` - This is how you can link a group to a specific Discord role. More info [here](#group-linking)<br/>
 
 ## Groups
-FlareBot doesn't go directly off roles, we instead use something called "groups", these groups can be given to a user and control which permissions people have. There is one group right off the bat which is called `Default` this is a group everyone has and every new person will get. If you wish to remove a permission from everyone then removing it from the `Default` group is the best way to go.<br/>
+FlareBot doesn't go directly off roles, we instead use something called "groups", these groups can be given to a user and control which permissions people have.<br/>
```
Comment (WalshyDev) [inline permissions.md]: Quick reference to negative perms for removing

File: permissions.md [hunk 5]
```diff
@@ -29,5 +37,21 @@ You can add a permission by simply doing the command `_perms group <name> add <p
...
-## Group linking
+### Group linking
 Group linking is basically linking a group to a role, this allows for you to add a role to a user __without__ them needing to have the group also added to them. It is a nice way to give a whole bunch of people certain permissions. Let's say we want to assign our `Muted` group to the `Muted` Discord role (which can be created by FlareBot in case of mutes). We can simply do `_perms group Muted link @Muted`, now here we can either mention the muted role or use the role ID (Which can be found in the `_roles` command).
+
+## Users
+Users can be given their own permissions individually from groups! These permissions will **always** overwrite group permissions so keep that in mind when checking issues with a permissions setup. Aside from their own permissions, users can also be assigned groups based on roles or manually adding the roles.
+
+For adding groups with roles simply link the group with a role (As shown above) and the user will have access to that groups permissions without any further actions!
+
+To add or remove groups from a user simply use `_perms user <user> group add|remove <group>` to add or remove a group from a user.
```
Comment (WalshyDev) [inline permissions.md:48]: This doesn't really make much sense

File: permissions.md [hunk 6]
```diff
@@ -29,5 +37,21 @@ You can add a permission by simply doing the command `_perms group <name> add <p
...
+
+### User permissions
+Adding and removing permissions is as simple as doing `_perms user <user> permission add|remove <perm>`! If you have trouble adding a permission, 
+remember that you can view all our permissions at any time by doing `_perms list`. To remove *all* permissions from a user you can do `_perms user <user> clear` at anytime!
```
Comment (WalshyDev) [inline permissions.md]: Why is this randomly on a new line when it's a continuation?

File: update.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+﻿# 4.2 the cleanup update
+## Changes
+ - Lots of back-end work (more info below)
+ - Permissions
+	 - The default group no longer exists.
```
Comment (WalshyDev) [inline update.md]: Removing the stupid amount of indenting, rename the file (said on Discord)

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: update.md (hunk 7 above)
Comment (weeryan17) [reply update.md]: It's literly a tab. whats wrong with using tabs for spaceing :(

File: permissions.md (hunk 5 above)
Comment (binaryoverload) [reply permissions.md:48]: Which part?

Comment (WalshyDev) [reply permissions.md:48]: It selected the wrong line - 46
This one doesn't have some annoying duplication in though, "to add or remove groups from a user... to ass or remove a group from a user"
So what does this do? Link right?

Comment (binaryoverload) [reply permissions.md:48]: Adding the roles links yeah

**Pull request merged**